### ISSUE TITLE
Runtime Error in 01_persist_supervised.ipynb

### ISSUE THREAD

**Issue by simomounir** [iancovert/persist#3]
Hi all,

I have come across this very promising gene marker selection and I want to try it on some example data that I have.

In the block where PERSIST is run, the following line:

**candidates, model = selector.eliminate(target=500, max_nepochs=250)**

returns a runtime Error. More details attached in the file.

[error.txt](https://github.com/iancovert/persist/files/11334037/error.txt)

Has anyone come across this bug? Could it be that it is trying to force us to use GPU?

Thanks in advance for looking into this.

**Comment by iancovert** [comment]
Hi Mohamed, thanks for checking out the package! I just looked at your error message and I don't think this is about the GPU (although I would highly recommend using one if you have access, it will make training much faster!). The problem appears to be a simple data type issue with your cell type labels. 

For PyTorch's cross entropy loss, it's required that the labels be of type `torch.long`, which is equivalent to int64. So when creating the `ExpressionDataset` object, if your second argument is currently called `labels`, could you try passing `labels.astype(np.int64)`? I think this may resolve the issue.

Would you also be able to let me know the shape of that argument, the `labels` variable? The `ExpressionDataset` object should have performed the conversion automatically, but I wonder if it didn't because the shape is perhaps `(n, 1)` rather than `(n,)`.

**Comment by simomounir** [comment]
Hi Ian,

Thanks for your prompt response. Appreciated.

I tried the example ipython notebook on github andI run into the same error I mentioned regarding the torch.long type.

I will go ahead with the solutions you suggested and see if that works.

**Comment by simomounir** [comment]
Hi Ian,

**labels.astype(np.int64)** passed as labels did not seem to work. Also, the shape of my labels' variable is:

**(10679,)**

I will see how to cast to "torch.long", although I already tried a couple ways that did not work.

Looking forward to hearing back from you.

Cheers.

**Comment by iancovert** [comment]
Hi there,

Apologies for the slow response. I tried running the notebook again to be sure, and I'm not encountering any issues - either with the notebook's original version or when I cast the labels before passing them using `labels.astype(np.int64)`. I also tried this when training on both GPU and CPU.

I'm not sure what could be causing the issue here. Which PyTorch version are you using? Maybe that could be an issue, perhaps it's a version that expects a different data type for the labels when using `torch.nn.CrossEntropyLoss`. I'm using 1.13.1, which is relatively up to date. 

Let me know.

Ian

**Comment by rhngla** [comment]
I was also unable to replicate this specific issue. Not sure if this was related, but `np.long` was [deprecated](https://numpy.org/devdocs/release/1.20.0-notes.html#using-the-aliases-of-builtin-types-like-np-int-is-deprecated) with `numpy 1.20+`. 

The `PERSIST` codebase and notebooks have now been updated; closing this issue since it couldn't be replicated on our side.
